src/extensions/default/CSSCodeHints/AtRuleCodeHints.js

```diff
@@ -0,0 +1,119 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, maxerr: 50 */
+/*global define, brackets, $ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    // Load dependent modules
+    var AppInit         = brackets.getModule("utils/AppInit"),
+        CodeHintManager = brackets.getModule("editor/CodeHintManager"),
+        AtRulesText     = require("text!AtRulesDef.json"),
+        AtRules         = JSON.parse(AtRulesText);
+
+
+    /**
+     * @constructor
+     */
+    function AtRuleHints() {
+    }
+
+    // As we are only going to provide @rules name hints
+    // we should claim that we don't have hints for anything else
+    AtRuleHints.prototype.hasHints = function (editor, implicitChar) {
+        var pos = editor.getCursorPos(),
+            token = editor._codeMirror.getTokenAt(pos),
+            cmState;
+
+        this.editor = editor;
+
+        if (token.state.base && token.state.base.localState) {
+            cmState = token.state.base.localState;
+        } else {
+            cmState = token.state;
+        }
+
+        // Check if we are at '@' rule 'def' context
+        if ((token.type === "def" && cmState.context.type === "at") || (token.type === "variable-2" && (cmState.context.type === "top" || cmState.context.type === "block"))) {
+            this.filter = token.string;
+            return true;
+        } else {
+            this.filter = null;
+            return false;
+        }
+    };
+
+    AtRuleHints.prototype.getHints = function (implicitChar) {
+        var pos     = this.editor.getCursorPos(),
+            token   = this.editor._codeMirror.getTokenAt(pos);
+
+        this.filter = token.string;
+        this.token = token;
+
+        if (this.filter) {
+
+            // Filter the property list based on the token string
+            var result = $.map(Object.keys(AtRules), function (key, value) {
+                    if (key.indexOf(token.string) === 0) {
+                        return key;
+                    }
+                }).sort();
+
+            return {
+                hints: result,
+                match: this.filter,
+                selectInitial: true,
+                defaultDescriptionWidth: true,
+                handleWideResults: false
+            };
+        } else {
+            return null;
+        }
+    };
+
+
+    /**
+     * Inserts a given @<rule> hint into the current editor context.
+     *
+     * @param {string} hint
+     * The hint to be inserted into the editor context.
+     *
+     * @return {boolean}
+     * Indicates whether the manager should follow hint insertion with an
+     * additional explicit hint request.
```

Comment:
We should use an uniform style to doc comments: the description should be always inline or always at new line.
I think the current style is on the same line.